ibis/pandas/core.py

```diff
@@ -384,3 +466,47 @@ def execute_and_reset(
     elif isinstance(result, pd.Series):
         return result.reset_index(drop=True)
     return result
+
+
+compute_time_context = Dispatcher(
+    'compute_time_context',
+    doc="""\
+
+Compute time context for a node in execution
+
+Notes
+-----
+For a given node, return with a list of timecontext that are going to be
+passed to its children nodes.
+time context is useful when data is not uniquely defined by op tree. e.g.
+a TableExpr can represent the query select count(a) from table, but the
+result of that is different with time context (20190101, 20200101) vs
+(20200101, 20210101), because what data is in "table" also depends on the
+time context. And such context may not be global for all nodes. Each node
+may have its own context. compute_time_context computes attributes that
+are going to be used in executeion and passes these attributes to children
+nodes.
+
+Param:
+timecontext : Optional[Tuple[pd.Timestamp, pd.Timestamp]]
+    begin and end time context needed for execution
+
+Return:
+List[Optional[Tuple[pd.Timestamp, pd.Timestamp]]]
+A list of timecontexts for children nodes of the current node. Note that
+timecontext are calculated for children nodes of computable args only.
+The length of the return list is same of the length of computable inputs.
+See ``computable_args`` in ``execute_until_in_scope``
+""",
+)
+
+
+@compute_time_context.register(ops.Node)
+def compute_time_context_default(node, timecontext, **kwargs):
+    return [timecontext for arg in node.inputs if is_computable_input(arg)]
+
+
+# In order to use time context feature, there must be a column of Timestamp
+# type, and named as 'time' in TableExpr. This TIME_COL constant will be
+# used in filtering data from a table or columns of a table.
+TIME_COL = 'time'
```

Comment:
where does this show up? can we allow this to be configurable?

Reply:
This column is used when table filter data, or operations like window trimming data in post-execution phase. We need to specify the column name for filtering.  I just leave it as it is as an assumption for now. Do you have any suggestions on how we should config it? Passing another argument in `execute()` seems too many args for the api.

Reply:
One possibility is to add to ibis.config

Reply:
yeah let's do that (can do this in a follown on)

Reply:
Got it, leave it as it is for now and will address this in a follow on.